huaweicloud/services/ga/resource_huaweicloud_ga_health_check.go

```diff
@@ -0,0 +1,530 @@
+
+import (
+	"context"
+	"fmt"
+	"regexp"
+	"strings"
+	"time"
+
+	"github.com/chnsz/golangsdk"
+
+	"github.com/hashicorp/go-multierror"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/common"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/config"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/utils"
+	"github.com/jmespath/go-jmespath"
+)
+
+func ResourceHealthCheck() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceHealthCheckCreate,
+		UpdateContext: resourceHealthCheckUpdate,
+		ReadContext:   resourceHealthCheckRead,
+		DeleteContext: resourceHealthCheckDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(10 * time.Minute),
+			Update: schema.DefaultTimeout(10 * time.Minute),
+			Delete: schema.DefaultTimeout(10 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+			"region": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+				ForceNew: true,
+			},
+			"enabled": {
+				Type:        schema.TypeBool,
+				Required:    true,
+				Description: `Specifies whether to enable health check.`,
+			},
+			"endpoint_group_id": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: `Specifies the endpoint group ID.`,
+				ValidateFunc: validation.All(
+					validation.StringMatch(regexp.MustCompile(`[a-fA-F0-9]{8}-[a-fA-F0-9]{4}-[a-fA-F0-9]{4}-[a-fA-F0-9]{4}-[a-fA-F0-9]{12}`),
+						"the input is invalid"),
+					validation.StringLenBetween(0, 36),
+				),
+			},
+			"interval": {
+				Type:         schema.TypeInt,
+				Required:     true,
+				Description:  `Specifies the health check interval, in seconds.`,
+				ValidateFunc: validation.IntBetween(1, 60),
+			},
+			"max_retries": {
+				Type:         schema.TypeInt,
+				Required:     true,
+				Description:  `Specifies the maximum number of retries.`,
+				ValidateFunc: validation.IntBetween(1, 10),
+			},
+			"port": {
+				Type:         schema.TypeInt,
+				Required:     true,
+				Description:  `Specifies the port used for the health check.`,
+				ValidateFunc: validation.IntBetween(1, 65535),
+			},
+			"timeout": {
+				Type:         schema.TypeInt,
+				Required:     true,
+				Description:  `Specifies the timeout duration of the health check, in seconds.`,
+				ValidateFunc: validation.IntBetween(1, 60),
+			},
+			"protocol": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Default:     "TCP",
+				Description: `Specifies the health check protocol.`,
+				ValidateFunc: validation.StringInSlice([]string{
+					"TCP",
```

Comment:
are you sure that only **TCP** are supportted?
If yes, please add a description in docs.

Reply:
fixed